misc/chrome_extension/content-script.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+var port = chrome.runtime.connect();
+//global since it needs to persist beyond the function scope
+var elementsToHide = [];
+
+//Parses the XPath
+//
+//sample return
+//{type:"title", value:[element, element]}
+//
+//title: The string for the type
+//xpathString: the xpath to parse
+//getTrue: BOOL, if the value return is the text instead of the element
+var parseXPath = function (title, xpathString, getText) {
+	try {
+		let nodes = document.evaluate(xpathString, document);
+		let e, elements = [];
+
+		if (nodes.resultType === XPathResult.UNORDERED_NODE_ITERATOR_TYPE || nodes.resultType === XPathResult.ORDERED_NODE_ITERATOR_TYPE) {
+			while (e = nodes.iterateNext()) {
+				if (getText) {
+					elements.push(e.nodeValue);
+				}
+				else {
+					elements.push(e);
+				}
+			}
+		}
+		else if (nodes.resultType === XPathResult.NUMBER_TYPE) {
+			elements.push(nodes.numberValue);
+		}
+		else if (nodes.resultType === XPathResult.STRING_TYPE) {
+			elements.push(nodes.stringValue);
+		}
+		else if (nodes.resultType === XPathResult.BOOLEAN_TYPE) {
+			elements.push(nodes.booleanValue);
+		}
+
+		return { type: title, value: elements };
+	}
+	catch (err) {
+		return { type: "__error", value: "Failed to parse XPath \'" + xpathString + "\'<br>Error: " + err };
+	}
+}
+
+//Parses the CSS selector
```

Comment:
Same comments as above for this function: 
- use jsdoc for comments
- change getTrue/getText for something like textOnly.